plugins/inputs/execd/execd_windows.go

```diff
@@ -19,10 +19,12 @@ func (e *Execd) Gather(acc telegraf.Accumulator) error {
 	switch e.Signal {
 	case "STDIN":
 		if osStdin, ok := e.process.Stdin.(*os.File); ok {
-			osStdin.SetWriteDeadline(time.Now().Add(1 * time.Second))
+			if err := osStdin.SetWriteDeadline(time.Now().Add(1 * time.Second)); err != nil {
+				return fmt.Errorf("setting write deadline failed: %w", err)
+			}
```

Comment:
How about ignoring the error if it says unsupported (i.e. [ErrNoDeadline](https://cs.opensource.google/go/go/+/master:src/os/error.go;bpv=1;bpt=1;l=26?q=%22file%20type%20does%20not%22&ss=go%2Fgo&gsn=ErrNoDeadline&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgo.googlesource.com%2Fgo%3Flang%3Dgo%3Fpath%3Dos%23var%2520ErrNoDeadline))? E.g.
```suggestion
			if err := osStdin.SetWriteDeadline(time.Now().Add(1 * time.Second)); err != nil {
				if !errors.Is(err, os.ErrNoDeadline) {
					return fmt.Errorf("setting write deadline failed: %w", err)
				}
			}
```

Reply:
Done